src/block-state/block-state.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { BlockState } from '@/block-state/block-state.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { TransactionsService } from '@/transactions/transactions.service';
+
+@Injectable()
+export class BlockStateService {
+    private cacheSize = 2016; // Define the maximum cache size
+
+    constructor(
+        @InjectRepository(BlockState)
+        private readonly blockStateRepository: Repository<BlockState>,
+        private readonly transactionService: TransactionsService,
+    ) {}
+
+    async getCurrentBlockState(): Promise<BlockState> {
+        const state = (
+            await this.blockStateRepository.find({
+                order: {
+                    indexedBlockHeight: 'DESC',
+                },
+                take: 1,
+            })
+        )[0];
+
+        return state;
+    }
+
+    async addBlockState(state: BlockState): Promise<void> {
+        this.blockStateRepository.save(state);
+        // Ensure the cache size does not exceed 2016
+        await this.trimState();
+    }
+
+    // Remove and return the latest item in the state cache
+    async dequeue_operation_state(): Promise<BlockState | null> {
+        const latest_state = (
+            await this.blockStateRepository.find({
+                order: {
+                    indexedBlockHeight: 'DESC',
+                },
+                take: 1,
+            })
+        )[0];
```

Comment:
```suggestion
        const latestState = await this.blockStateRepository.findOne({
            order: {
                indexedBlockHeight: 'DESC',
            },
        });
```

Reply:
does not work because selection statement is not added.